src/replication.c

```diff
@@ -3487,6 +3521,48 @@ void waitCommand(client *c) {
     replicationRequestAckFromSlaves();
 }
 
+/* WAIT for N replicas and / or local master to acknowledge our latest
+ * write command got synced to the disk. */
+void waitaofCommand(client *c) {
+    mstime_t timeout;
+    long numreplicas, numlocal, ackreplicas, acklocal;
+
+    /* Argument parsing. */
+    if (getRangeLongFromObjectOrReply(c,c->argv[1],0,1,&numlocal,NULL) != C_OK)
```

Comment:
Why is it named "numlocal"? it's actually bool and not num.

Reply:
i'm not certain about it.
it seems easier to be consistent with `numreplicas` and `ackreplicas` and use `>=` for the checks.

maybe the discussion we should really be having here is about the command syntax (arguments and reply).
the argument could have been a keyword, e.g. "local" being present or missing, and the respond could be a bool.
i felt it is better to keep it an integer that's always either 0 or 1.

p.s. maybe in some future design we can have more than one local copy of an AOF file (possibly "local" means network attached storage), and there would be some reason to ask that it's persisted to both. :man_shrugging:

Reply:
This raises again the question about dropping the differentiation between master and replicas and simply referring to the total number of instances that have fsynced. It's more generalized and could leave us more room for implementation changes in the future.

Reply:
i'm afraid it could turn out to be non-useful (for failover triggering concerns). i think the trading of boolean for integer actually leaves us with some flexibility, but it's hard to predict which one of these we'll regret.